### PR TITLE
Microsoft defender for endpoints

### PR DESCRIPTION
**Pull request by squioc**
add new intake for Microsoft defender for endpoints

### CODE REVIEW

File: Microsoft Defender/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Microsoft Defender
+
+## Description
+Microsoft Defender is an offer from Microsoft for individuals, business and enterprise to protect their digital assets (endpoints, accounts, emails, ...)
```
Comment (PierrickV) [inline Microsoft Defender/README.md:4]:
```suggestion
Microsoft Defender for Endpoint is an enterprise endpoint security platform designed to help enterprise networks prevent, detect, investigate, and respond to advanced threats.
```

I suggest that we use the sentence from [docs.microsoft.com](https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/microsoft-defender-endpoint?view=o365-worldwide)

File: Microsoft Defender/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+## Intakes
+*
```
Comment (PierrickV) [inline Microsoft Defender/README.md:7]:
```suggestion
```

Empty section

File: Microsoft Defender/_meta/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+uuid: 8d4be57f-9444-4f1a-acf7-670b6e2b5ba5
+name: Microsoft Defender
+slug: microsoft-defender
+description: >-
+  Microsoft Defender is an offer from Microsoft for individuals, business and enterprise to protect their digital assets (endpoints, accounts, emails, ...)
```
Comment (PierrickV) [inline Microsoft Defender/_meta/manifest.yml:5]:
```suggestion
  Microsoft Defender for Endpoint is an enterprise endpoint security platform designed to help enterprise networks prevent, detect, investigate, and respond to advanced threats.
```

File: Microsoft Defender/microsoft-defender-for-endpoints/_meta/manifest.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+uuid: 05e6f36d-cee0-4f06-b575-9e43af779f9f
+name: Microsoft Defender for Endpoints
+slug: microsoft-defender-for-endpoints
+
+description: >-
+  Microsoft Defender for Endpoints protects endpoints of our enterprise against external threats
```
Comment (PierrickV) [inline Microsoft Defender/microsoft-defender-for-endpoints/_meta/manifest.yml:6]:
```suggestion
  Microsoft Defender for Endpoint is an enterprise endpoint security platform designed to help enterprise networks prevent, detect, investigate, and respond to advanced threats.
```

File: Microsoft Defender/microsoft-defender-for-endpoints/_meta/smart-descriptions.json [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      },
+      {
+        "source": "file.x509.serial_number",
+        "target": "microsoft.defender.certificate.signer",
+        "type": "signed by"
+      }
+    ]
+  },
+  {
+    "value": "New process {action.properties.InitiatingProcessFileName} on {host.name}",
```
Comment (PierrickV) [inline Microsoft Defender/microsoft-defender-for-endpoints/_meta/smart-descriptions.json:49]:
```suggestion
    "value": "New process {action.properties.InitiatingProcessFileName} created on {host.name}",
```

File: Microsoft Defender/microsoft-defender-for-endpoints/_meta/smart-descriptions.json [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    "relationships": [
+      {
+        "source": "host.name",
+        "target": "action.properties.InitiatingProcessFileName",
+        "type": "start process"
+      }
+    ]
+  },
+  {
+    "value": "{user.name} logged on {host.name}",
```
Comment (PierrickV) [inline Microsoft Defender/microsoft-defender-for-endpoints/_meta/smart-descriptions.json:71]: Could you add the domain ? example `WORKGROUP\squioc` `foo.corp\squioc` ?

File: Microsoft Defender/microsoft-defender-for-endpoints/_meta/smart-descriptions.json [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    "relationships": [
+      {
+        "source": "user.name",
+        "target": "host.name",
+        "type": "logged on"
+      }
+    ]
+  },
+  {
+    "value": "{source.address} connect to {destination.address}",
```
Comment (PierrickV) [inline Microsoft Defender/microsoft-defender-for-endpoints/_meta/smart-descriptions.json:93]:
```suggestion
    "value": "{source.address} connected to {destination.address}",
```

**Comment by PierrickV** [comment]
We might be missing few smart descriptions as there is only 4 for a very big `parser.yml` file (385 lines).

I think that this work should be merged and tested which will make easier to find missing smart descriptions.

**Review by PierrickV — DISMISSED** [review]

**Comment by PierrickV** [comment]
Looks like your work is missing [Event Categorization Fields](https://www.elastic.co/guide/en/ecs/current/ecs-category-field-values-reference.html) such as `event.category`

File: Microsoft Defender/microsoft-defender-for-endpoints/tests/test_device_event.json [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    },
+    "host": {
+      "id": "1111111111111111111111111111111111111111",
+      "name": "test.lab"
+    },
+    "process": {
+      "pid": 1664,
+      "start": "2022-09-01T06:56:23.7887846Z",
+      "executable": "software_reporter_tool.exe",
+      "command_line": "\"software_reporter_tool.exe\" --use-crash-handler-with-id=\"\\\\.\\pipe\\crashpad_11111_XXXXXXXXXXXXXXXX\" --sandboxed-process-id=2 --init-done-notifier=804 --sandbox-mojo-pipe-token=********** --mojo-platform-channel-handle=780 --engine=2",
```
Comment (PierrickV) [inline Microsoft Defender/microsoft-defender-for-endpoints/tests/test_device_event.json:33]: Seems not usable for detection, maybe the escape chars come from your tests ?

**Comment by PierrickV** [comment]
It would be nice to use the timestamp of the event to set the '@timestamp' field, what do you think ?

File: Microsoft Defender/microsoft-defender-for-endpoints/tests/test_device_event.json (hunk 8 above)
Comment (squioc) [reply Microsoft Defender/microsoft-defender-for-endpoints/tests/test_device_event.json:34]: The escaped chars are only for the test. For the detection, you will have the string:
```
"software_reporter_tool.exe" --use-crash-handler-with-id="\\.\pipe\crashpad_11111_XXXXXXXXXXXXXXXX" --sandboxed-process-id=2 --init-done-notifier=804 --sandbox-mojo-pipe-token=********** --mojo-platform-channel-handle=780 --engine=2
```

File: Microsoft Defender/microsoft-defender-for-endpoints/_meta/manifest.yml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  Loaded DLLs: to be defined
+  Named Pipes: to be defined
+  PowerShell logs: to be defined
+  Process command-line parameters: to be defined
+  Process monitoring: to be defined
+  Process use of network: to be defined
+  Services: to be defined
+  Windows event logs: to be defined
+  Windows Registry: to be defined
+  WMI Objects: to be defined
```
Comment (PierrickV) [inline Microsoft Defender/microsoft-defender-for-endpoints/_meta/manifest.yml:23]: :face_in_clouds:

File: Microsoft Defender/microsoft-defender-for-endpoints/_meta/smart-descriptions.json [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+        "field": "action.type",
+        "value": "ScheduledTaskUpdated"
+      },
+      {
+        "field": "host.name"
+      }
+    ]
+  },
+  {
+    "value": "{microsoft.defender.observer.interface.type} interface {microsoft.defender.observer.interface.name} on {host.name} is {microsoft.defender.observer.interface.status}",
```
Comment (PierrickV) [inline Microsoft Defender/microsoft-defender-for-endpoints/_meta/smart-descriptions.json:152]: I think that if you use this kind of fields (non ECS) in a smart description, you should modify your entries in `fields.yml` 

Example
https://github.com/SEKOIA-IO/intake-formats/blob/8cb0df409da2c1a313f86c54c8e401e8e2051022/Windows/windows/_meta/fields.yml#L1-L8

File: Microsoft Defender/microsoft-defender-for-endpoints/_meta/manifest.yml (hunk 9 above)
Comment (squioc) [reply Microsoft Defender/microsoft-defender-for-endpoints/_meta/manifest.yml]: Done

File: Microsoft Defender/microsoft-defender-for-endpoints/_meta/smart-descriptions.json (hunk 10 above)
Comment (squioc) [reply Microsoft Defender/microsoft-defender-for-endpoints/_meta/smart-descriptions.json:152]: I checked, and it seems that it's not necessary.

**Review by PierrickV — APPROVED** [review]

**Pull request merged**